go/kbfs/simplefs/simplefs.go

```diff
@@ -3022,3 +3010,55 @@ func (k *SimpleFS) SimpleFSConfigureDownload(
 	k.downloadManager.configureDownload(arg.CacheDirOverride, arg.DownloadDirOverride)
 	return nil
 }
+
+// SimpleFSGetGUIFileContext implements the SimpleFSInterface.
+func (k *SimpleFS) SimpleFSGetGUIFileContext(ctx context.Context,
+	path keybase1.KBFSPath) (resource keybase1.GUIFileContext, err error) {
+	if len(path.Path) == 0 {
+		return keybase1.GUIFileContext{}, errors.New("empty path")
+	}
+
+	if k.localHTTPServer == nil {
+		return keybase1.GUIFileContext{}, errors.New("HTTP server is disabled")
+	}
+	// Refresh the token every time. This RPC is called everytime a file is
+	// being viewed and we have a cache size of 64 so this shouldn't be a
+	// problem.
+	token, err := k.localHTTPServer.NewToken()
+	if err != nil {
+		return keybase1.GUIFileContext{}, err
+	}
+	address, err := k.localHTTPServer.Address()
+	if err != nil {
+		return keybase1.GUIFileContext{}, err
+	}
+
+	var encodedSegments []string
+	for _, segment := range strings.Split(path.Path, "/") {
+		encodedSegments = append(encodedSegments, url.PathEscape(segment))
+	}
+
+	u := url.URL{
+		Scheme:   "http",
+		Host:     address,
+		Path:     "/files" + strings.Join(encodedSegments, "/"),
```

Comment:
Is this assuming that `path.Path` starts with `/`?  Why not do `strings.Join` over the whole thing including "/files"?